File: dayflow-hrms/frontend/src/services/leaveService.js
import api from './api';

export const leaveService = {
  applyLeave: (data) => api.post('/leaves', data),
  getMyLeaves: () => api.get('/leaves/me'),
  getLeaveById: (id) => api.get(`/leaves/${id}`),
  updatePendingLeave: (id, data) => api.put(`/leaves/${id}`, data),
  withdrawPendingLeave: (id) => api.post(`/leaves/${id}/withdraw`),
  getAllLeaves: (status) => api.get('/admin/leaves', { params: { status } }),
  approveLeave: (id, data) => api.post(`/admin/leaves/${id}/approve`, data || {}),
  rejectLeave: (id, data) => api.post(`/admin/leaves/${id}/reject`, data || {})
};

File: dayflow-hrms/backend/src/main/java/com/dayflow/hrms/service/LeaveService.java
package com.dayflow.hrms.service;

import com.dayflow.hrms.dto.leave.*;
import com.dayflow.hrms.enums.LeaveStatus;

import java.util.List;

public interface LeaveService {
    LeaveDto applyLeave(Long userId, LeaveApplicationRequest request, String ipAddress);
    LeaveDto updatePendingLeave(Long leaveId, Long userId, LeaveUpdateRequest request, String ipAddress);
    LeaveDto withdrawPendingLeave(Long leaveId, Long userId, String ipAddress);
    LeaveDto approveLeave(Long leaveId, LeaveReviewRequest request, Long adminUserId, String ipAddress);
    LeaveDto rejectLeave(Long leaveId, LeaveReviewRequest request, Long adminUserId, String ipAddress);
    List<LeaveDto> getMyLeaves(Long userId);
    List<LeaveDto> getAllLeaves(LeaveStatus statusFilter);
    LeaveDto getLeaveById(Long leaveId);
}
